homeassistant/components/zwave_js/sensor.py

```diff
@@ -509,19 +509,31 @@ def __init__(
         # Entity class attributes
         self._attr_name = self.generate_name(include_value_name=True)
 
+    @property
+    def device_class(self) -> SensorDeviceClass | None:
+        """Return sensor device class."""
+        if super().device_class is not None:
+            return super().device_class
+        if self.info.primary_value.metadata.states:
+            return SensorDeviceClass.ENUM
```

Comment:
The state must be non numeric. Is that the case? The class docstring indicates otherwise.